recharges/tasks.py

```diff
@@ -252,3 +252,79 @@ def run(self, recharge_id, **kwargs):
 
 
 hotsocket_get_airtime = Hotsocket_Get_Airtime()
+
+
+class Check_Hotsocket_Status(Task):
+
+    """
+    Task to check hotsocket recharge request and sets the recharge model
+    status to successful if the airtime has been loaded to the user's phone.
+    """
+    name = "recharges.tasks.Check_Hotsocket_Status"
+
+    def prep_hotsocket_status_dict(self, recharge_id):
+        """
+        Constructs the dict needed to make a hotsocket recharge status request
+        """
+
+        hotsocket_data = {
+            'username': settings.HOTSOCKET_API_USERNAME,
+            'as_json': True,
+            'token': get_token(),
+            'reference': recharge_id + int(settings.HOTSOCKET_REFBASE),
+        }
+        return hotsocket_data
+
+    def request_hotsocket_status(self, recharge_id):
+        hotsocket_data = self.prep_hotsocket_status_dict(recharge_id)
+        recharge_status_post = requests.post("%s/status" %
+                                             settings.HOTSOCKET_API_ENDPOINT,
+                                             data=hotsocket_data)
+        return recharge_status_post.json()
+
+    def run(self, recharge_id, **kwargs):
+        l = self.get_logger(**kwargs)
+        l.info("Looking up Hotsocket status")
+        hs_status = self.request_hotsocket_status(recharge_id)
+        hs_status_code = hs_status["response"]["status"]
+
+        if hs_status_code == "0000":
+            # recharge successful
```

Comment:
this just means lookup was successful - not the actual recharge - see section 11.4 of the guide